plugins/exporter.koplugin/target/markdown.lua

```diff
@@ -0,0 +1,160 @@
+    self:populateSettings()
+    return {
+        text = _("Markdown"),
+        checked_func = function() return self:isEnabled() end,
+        sub_item_table = {
+            {
+                text = _("Export to Markdown"),
+                checked_func = function() return self:isEnabled() end,
+                callback = function() self:toggleEnabled() end,
+            },
+            {
+                text = _("Format Highlights based on style"),
+                checked_func = function() return self.settings.highlight_formatting end,
+                callback = function() self.settings.highlight_formatting = not self.settings.highlight_formatting end,
+            },
+            {
+                text = _("Lighten"),
+                keep_menu_open = true,
+                callback = function()
+                    self:editFormatStyle("lighten")
+                end,
+            },
+            {
+                text = _("Strikeout"),
+                keep_menu_open = true,
+                callback = function()
+                    self:editFormatStyle("strikeout")
+                end,
+            },
+            {
+                text = _("Underscore"),
+                keep_menu_open = true,
+                callback = function()
+                    self:editFormatStyle("underscore")
+                end,
+            },
+            {
+                text = _("Invert"),
+                keep_menu_open = true,
+                callback = function()
+                    self:editFormatStyle("invert")
+                end,
+            },
+        }
+    }
+end
+
+function MarkdownExporter:export(t)
+    self:populateSettings()
+    local path = self:getFilePath(t)
+    local file = io.open(path, "w")
+    if not file then return false end
+    for idx, book in ipairs(t) do
+        local current_chapter = nil
+        file:write("# " .. book.title .. "\n")
+        file:write("##### " .. book.author:gsub("\n", ", ") .. "\n\n")
+        for _, note in ipairs(book) do
+            local entry = note[1]
+            if entry.chapter ~= current_chapter then
+                current_chapter = entry.chapter
+                file:write("## " .. current_chapter .. "\n")
+            end
+            file:write("### " .. os.date("%d %B %Y %I:%M %p", entry.time) .. ", Page " .. entry.page .. "\n")
+            if self.settings.highlight_formatting then
+                file:write(string.format(formatters[self.settings.formatting_options[entry.drawer]], entry.text) .."\n")
+            else
+                file:write(entry.text .. "\n")
+            end
+            if entry.note then
+                file:write("\n---\n" .. entry.note .. "\n")
+            end
+            file:write("\n")
+        end
+        if idx < #t then
+            file:write("\n")
+        end
+    end
+    file:write("\n\n_Generated at: " .. self:getTimeStamp() .. "_")
+    file:close()
+    return true
+end
+
+return MarkdownExporter
```

Comment:
At the end there should be a newline. (Sorry overlooked that first.)

Reply:
You have not. I removed it.

Added it back again.